gnocchi/rest/auth_helper.py

```diff
@@ -118,8 +118,10 @@ def get_metric_policy_filter(request, rule):
 class BasicAuthHelper(object):
     @staticmethod
     def get_current_user(request):
-        auth = werkzeug.http.parse_authorization_header(
-            request.headers.get("Authorization"))
+        hdr = request.headers.get("Authorization")
+        auth_hdr = (hdr.decode('utf-8') if isinstance(hdr, bytes)
+                    else hdr)
+        auth = werkzeug.http.parse_authorization_header(auth_hdr)
```

Comment:
This is unrelated to SQLAlchemy. So maybe write why you need this somewhere?

Reply:
This is Python3 related, I tested this with Python 3.8 as in the PR title. Without the change you get the following when running ``py38-mysql-file``.

```
======
Totals
======
Ran: 1103 tests in 171.5752 sec.
 - Passed: 1006
 - Skipped: 25
 - Expected Fail: 2
 - Unexpected Success: 0
 - Failed: 70
```

All errors is:

```
Captured traceback:
~~~~~~~~~~~~~~~~~~~
    b'Traceback (most recent call last):'
    b'  File "/home/ubuntu/gnocchi/gnocchi/tests/base.py", line 60, in skip_if_not_implemented'
    b'    return func(*args, **kwargs)'
    b'  File "/home/ubuntu/gnocchi/gnocchi/tests/test_rest.py", line 224, in test_status'
    b'    r = self.app.get("/v1/status")'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webtest/app.py", line 327, in get'
    b'    return self.do_request(req, status=status,'
    b'  File "/home/ubuntu/gnocchi/gnocchi/tests/test_rest.py", line 128, in do_request'
    b'    response = super(TestingApp, self).do_request(req, *args, **kwargs)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webtest/app.py", line 623, in do_request'
    b'    res = req.get_response(app, catch_exc_info=True)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/request.py", line 1309, in send'
    b'    status, headers, app_iter, exc_info = self.call_application('
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/request.py", line 1278, in call_application'
    b'    app_iter = application(self.environ, start_response)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webtest/lint.py", line 201, in lint_app'
    b'    iterator = application(environ, start_response_wrapper)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/dec.py", line 129, in __call__'
    b'    resp = self.call_func(req, *args, **kw)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/dec.py", line 193, in call_func'
    b'    return self.func(req, *args, **kwargs)'
    b'  File "/home/ubuntu/gnocchi/gnocchi/rest/http_proxy_to_wsgi.py", line 60, in __call__'
    b'    return req.get_response(self.application)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/request.py", line 1313, in send'
    b'    status, headers, app_iter = self.call_application('
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/request.py", line 1278, in call_application'
    b'    app_iter = application(self.environ, start_response)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/dec.py", line 129, in __call__'
    b'    resp = self.call_func(req, *args, **kw)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/dec.py", line 193, in call_func'
    b'    return self.func(req, *args, **kwargs)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/oslo_middleware/base.py", line 124, in __call__'
    b'    response = req.get_response(self.application)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/request.py", line 1313, in send'
    b'    status, headers, app_iter = self.call_application('
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/webob/request.py", line 1278, in call_application'
    b'    app_iter = application(self.environ, start_response)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/paste/urlmap.py", line 216, in __call__'
    b'    return app(environ, start_response)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/pecan/middleware/recursive.py", line 56, in __call__'
    b'    return self.application(environ, start_response)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/pecan/core.py", line 852, in __call__'
    b'    return super(Pecan, self).__call__(environ, start_response)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/pecan/core.py", line 693, in __call__'
    b'    self.invoke_controller(controller, args, kwargs, state)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/pecan/core.py", line 584, in invoke_controller'
    b'    result = controller(*args, **kwargs)'
    b'  File "/home/ubuntu/gnocchi/gnocchi/rest/api.py", line 2143, in get'
    b'    enforce("get status", {})'
    b'  File "/home/ubuntu/gnocchi/gnocchi/rest/api.py", line 109, in enforce'
    b'    creds = pecan.request.auth_helper.get_auth_info(pecan.request)'
    b'  File "/home/ubuntu/gnocchi/gnocchi/rest/auth_helper.py", line 132, in get_auth_info'
    b'    user = self.get_current_user(request)'
    b'  File "/home/ubuntu/gnocchi/gnocchi/rest/auth_helper.py", line 125, in get_current_user'
    b'    auth = werkzeug.http.parse_authorization_header('
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/werkzeug/http.py", line 666, in parse_authorization_header'
    b'    value = _wsgi_decoding_dance(value)'
    b'  File "/home/ubuntu/gnocchi/.tox/py38-mysql-file/lib/python3.8/site-packages/werkzeug/_internal.py", line 153, in _wsgi_decoding_dance'
    b'    return s.encode("latin1").decode(charset, errors)'
    b"AttributeError: 'bytes' object has no attribute 'encode'"
```